synfig-studio/src/gui/widgets/widget_timeslider.cpp

```diff
@@ -349,21 +387,58 @@ bool
 Widget_Timeslider::on_button_release_event(GdkEventButton *event){
 	SYNFIG_EXCEPTION_GUARD_BEGIN()
 	lastx = (double)event->x;
+	move_lower_bound_button = false;
+	move_upper_bound_button = false;
 	return event->button == 1 || event->button == 2;
 	SYNFIG_EXCEPTION_GUARD_END_BOOL(true)
 }
 
+void
+Widget_Timeslider::get_bounds_rectangle_dimensions(double& x0, double& x1, double& w , bool lower)
+{
+	if(lower){
+		 x0 = time_plot_data->get_double_pixel_t_coord(time_plot_data->lower_ex);
+		 x1 = time_plot_data->get_double_pixel_t_coord(time_plot_data->time_model->get_play_bounds_lower());
+		 w = x1 - x0;
+	}
+	else {
+		 x0= time_plot_data->get_double_pixel_t_coord(time_plot_data->time_model->get_play_bounds_upper());
+		 x1= time_plot_data->get_double_pixel_t_coord(time_plot_data->upper_ex);
+		 w= x1 - x0;
+	}
+}
+
 bool
 Widget_Timeslider::on_motion_notify_event(GdkEventMotion* event) //for dragging
 {
 	SYNFIG_EXCEPTION_GUARD_BEGIN()
+
+	double current_x = event->x;
+	double x0,x1,w,z0,z1,width;
+	get_bounds_rectangle_dimensions(x0,x1,w,true);
+	get_bounds_rectangle_dimensions(z0,z1,width,false);
+
+	if(move_lower_bound_button || (current_x <= x1 && current_x >= x1 - boundary_dimension))
+		   get_window()->set_cursor(bounds_cursor);
+	else if(move_upper_bound_button || ((current_x <= (z0 + boundary_dimension))  && (current_x >= (z0))))
+		   get_window()->set_cursor(bounds_cursor);
+	else
+		   get_window()->set_cursor(default_cursor);
+
 	double dx = (double)event->x - lastx;
 	lastx = (double)event->x;
 
 	if (!time_plot_data->time_model || get_width() <= 0 || get_height() <= 0)
 		return false;
 
-	Gdk::ModifierType mod = Gdk::ModifierType(event->state);
+	Gdk::ModifierType mod = Gdk::ModifierType(event->state);//mouse buttons are also modifiers
```

Comment:
remove comment?

Reply:
oh yes definitely I just comment on random things when I first start working on the feature to understand stuff, and sometimes I forget to delete some of them xD.